cmd/kusk/cmd/root.go

```diff
@@ -59,53 +59,21 @@ var rootCmd = &cobra.Command{
 	Use:   "kusk",
 	Short: "",
 	Long:  ``,
-	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
 		analytics.SendAnonymousCMDInfo(nil)
 
-		if isatty.IsTerminal(os.Stdout.Fd()) == true && build.Version != "latest" {
+		versionCheck(cmd)
 
-			if len(build.Version) != 0 {
-				ghclient, err := utils.NewGithubClient("", nil)
-				if err != nil {
-					errors.NewErrorReporter(cmd, err).Report()
-					return
-				}
-
-				ref, err := ghclient.GetLatest(kuskgateway)
-				if err != nil {
-					errors.NewErrorReporter(cmd, err).Report()
-					return
-				}
-
-				latestVersion, err := version.NewVersion(ref)
-				if err != nil {
-					errors.NewErrorReporter(cmd, err).Report()
-					return
-				}
-
-				currentVersion, err := version.NewVersion(build.Version)
-				if err != nil {
-					errors.NewErrorReporter(cmd, err).Report()
-					return
-				}
-
-				if currentVersion != nil && currentVersion.LessThan(latestVersion) {
-					kuskui.PrintWarning(fmt.Sprintf("This version %s of Kusk cli is outdated. The latest version available is %s", currentVersion, latestVersion))
-
-					if runtime.GOOS == "windows" {
-						kuskui.PrintWarning(fmt.Sprintf("Run the following command to update Kusk CLI. \n\n go install -x github.com/kubeshop/kusk-gateway/cmd/kusk@latest	\n kusk cluster upgrade\n"))
-					}
-					if runtime.GOOS == "linux" {
-						kuskui.PrintWarning(fmt.Sprintf("Run the following command to update Kusk CLI. \n\n curl -sSLf https://raw.githubusercontent.com/kubeshop/kusk-gateway/main/cmd/kusk/scripts/install.sh | bash \n kusk cluster upgrade\n"))
-					}
-					if runtime.GOOS == "darwin" {
-						kuskui.PrintWarning(fmt.Sprintf("Run the following command to update Kusk CLI. \n\n brew install kubeshop/kusk/kusk \n kusk cluster upgrade\n"))
-					}
+		if cmd.Use == "generate" || cmd.Use == "deploy" {
```

Comment:
@kylehodgetts this is an attempt to clean up our code a little

Reply:
This is fine for now but generally I'd like to avoid the root command knowing which of its child commands was called, it increases coupling

Reply:
Ideally parent command should do this